### ISSUE TITLE
Change `Bytes` from `string` alias to avoid misuse

### ISSUE THREAD

**Issue by austinabell** [near/near-sdk-js#117]
Somewhat related to #78 but the issue currently is that someone can pass any string into any function that accepts `Bytes`. This is a foot gun because it will cause bugs.

Uncertain if the jsvm panics if it encounters a value >8bits (since strings are utf16), but regardless it causes a crash at runtime or UB to the developer.

Intuitively, and this might not be ideal but to take a guess, `Bytes` would be a class that can be constructed by multiple types but validated that it's bytes and maybe the inputs are `Bytes | Uint8Array` since `Uint8Array` seems like it would be a common type from libraries.

Alternatively only accepting `Bytes` and forcing developers to be more explicit with what they pass in. Likely https://github.com/near/near-sdk-js/blob/37965b8b1f5f467761938229fedf9cd0532cfd24/src/utils.js#L17-L24 would be the logic used by the constructor of bytes.

**Comment by ailisp** [comment]
You're right, it's too easy to misuse. The way you suggested can be a good way. One problem is the logic of `bytes` is expensive for very long bytes, such as when represent factory contract of another JS contract, in this case we would want to skip the check, or make the build pipeline optimized Bytes constructor of constant bytes to compile-time check.

> Uncertain if the jsvm panics if it encounters a value >8bits (since strings are utf16), but regardless it causes a crash at runtime or UB to the developer.

The result of doing so is incorrect but safe, as raw bytes of the string would be used in C, and passed to vm logic. It is confirmed by this [test](https://github.com/near/near-sdk-js/pull/143/files#diff-7b5adb785fa2951909fec70d54b8941c83b3117cb7f89cb8347af8698d238caf) and you can also reason about this behavior in this [function](https://github.com/near/quickjs/pull/1/files#diff-45f1ae674139f993bf8a99c382c1ba4863272a6fec2f492d76d7ff1b2cfcfbe2R3963)

**Comment by ailisp** [comment]
Fully addressed in #308
